Fix E1102 / ``not-callable`` false positive for property that returns a lambda function conditionally (#6068)
PyCQA/pylint: doc/whatsnew/2.14.rst

Other Changes
=============

> * Only raise ``not-callable`` when all the inferred values of an expression are not callable.

Reviewer: ```suggestion
* Only raise ``not-callable`` when all the inferred values of a property are not callable.
```
